app/controllers/posts_controller.rb

```diff
@@ -4,7 +4,7 @@ class PostsController < ApplicationController
   before_action :require_login, except: [:index, :show]
 
   def index
-    @posts = @user.posts.all
+    @posts = @user.posts.order("created_at").page(params[:page]).per(20)
```

Comment:
Change `20` to `PER_PAGE`.

Then, create an initializer in config/initializers along the lines of:

```
PER_PAGE = Rails.env.test? ? 5 : 20
```